### PR TITLE
Swift 4.2

### PR DESCRIPTION
**Pull request by nathantannar4**
What does this implement/fix? Explain your changes.
---------------------------------------------------
Updates the codebase to Swift 4.2, specifies Swift 4.2 version of `MessageInputBar`, update CircleCi config

Does this close any currently open issues?
------------------------------------------
Everyone requesting Swift 4.2 😊 

Any other comments?
-------------------
Sorry to those who have submitted Swift 4.2 PRs already. #880 was branched from `master`, not `development` and we have already made changes in `development`

For those who want early access, use the branch `swift-4.2`

This will segue `MessageKit` to its 1.1 release

### CODE REVIEW

**Review by SD10 — APPROVED** [review]
@nathantannar4 Please add a CHANGELOG entry for the upcoming release - "Added Swift 4.2 support"

File: Sources/Views/PlayButtonView.swift [hunk 1]
```diff
@@ -28,7 +28,7 @@ open class PlayButtonView: UIView {
 
     // MARK: - Properties
 
-    open let triangleView = UIView()
+    public let triangleView = UIView()
```
Comment (zhongwuzw) [inline Sources/Views/PlayButtonView.swift:31]: Emm, how about make it `open` and `var`?

Comment (nathantannar4) [reply Sources/Views/PlayButtonView.swift:31]: What would be the use case for allowing someone to set a new value?

Comment (zhongwuzw) [reply Sources/Views/PlayButtonView.swift:31]: Maybe users want to do some custom things? Make it `open` have no side-effect, right?

Comment (nathantannar4) [reply Sources/Views/PlayButtonView.swift:31]: Before it was open but a let. So technically they can't change it. But if it's changed to var someone can set it to something. But if they set it that won't automatically add the new view to the superview or lay it out.

**Comment by hyouuu** [comment]
This is awesome! Please update MessageInputBar as well :D

**Comment by hyouuu** [comment]
Oh I see the repo is updated, but cocoapod didn't update it - probably because the podspec didn't specify a version. I had to add MessageInputBar directly in my pod file, then it was updated from 0.2.2 to 0.4.0

**Comment by talanov** [comment]
@nathantannar4 PodSpec invalid, MessageKit's InputBar automatically install 0.2.2 version instead of 0.4.

**Comment by nathantannar4** [comment]
@talanov I didn't officially released a new pod version for MessageInputBar bar since this is wasn't a release yet. You will need to specify the version until official release. 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by SD10** [comment]
@nathantannar4 You can release `MessageInputBar 0.4.0` though? It shouldn't be blocked by a MessageKit release

**Comment by nathantannar4** [comment]
Sure, released just now

**Pull request merged**